FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,942 @@
+        if let prefix = prefix, !prefix.isEmpty {
+          keys = Set(config.keys.filter { $0.hasPrefix(prefix) })
+        } else {
+          keys = Set(config.keys)
+        }
+      }
+    }
+    return keys
+  }
+
+  public func countByEnumerating(with state: UnsafeMutablePointer<NSFastEnumerationState>,
+                                 objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>,
+                                 count len: Int) -> Int {
+    var count = 0
+    queue.sync {
+      let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+
+      if let config = configContent.activeConfig()[fullyQualifiedNamespace] as? NSDictionary {
+        count = config.countByEnumerating(with: state, objects: buffer, count: len)
+      }
+    }
+    return count
+  }
+
+  // MARK: Defaults
+
+  /// Sets config defaults for parameter keys and values in the default namespace config.
+  /// - Parameter defaults         A dictionary mapping a NSString * key to a NSObject * value.
+  @objc public func setDefaults(_ defaults: [String: Any]?) {
+    let defaults = defaults ?? [String: Any]()
+    let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+    queue.async { [weak self] in
+      guard let self = self else { return }
+
+      self.configContent.copy(fromDictionary: [fullyQualifiedNamespace: defaults],
+                              toSource: .default,
+                              forNamespace: fullyQualifiedNamespace)
+      self.settings.lastSetDefaultsTimeInterval = Date().timeIntervalSince1970
+    }
+  }
+
+  /// Sets default configs from plist for default namespace.
+  ///
+  /// - Parameter fileName The plist file name, with no file name extension. For example, if the
+  /// plist
+  /// file is named `defaultSamples.plist`:
+  ///                 `RemoteConfig.remoteConfig().setDefaults(fromPlist: "defaultSamples")`
+  @objc(setDefaultsFromPlistFileName:)
+  public func setDefaults(fromPlist fileName: String?) {
+    guard let fileName = fileName, !fileName.isEmpty else {
+      RCLog.warning("I-RCN000037",
+                    "The plist file name cannot be nil or empty.")
+      return
+    }
+
+    for bundle in [Bundle.main, Bundle(for: type(of: self))] {
+      if let path = bundle.path(forResource: fileName, ofType: "plist"),
+         let config = NSDictionary(contentsOfFile: path) as? [String: Any] {
+        setDefaults(config)
+        return
+      }
+    }
+    RCLog.warning("I-RCN000037",
+                  "The plist file '\(fileName)' could not be found by Remote Config.")
+  }
+
+  /// Returns the default value of a given key from the default config.
+  ///
+  /// - Parameter key              The parameter key of default config.
+  /// - Returns                 Returns the default value of the specified key. Returns
+  ///                         nil if the key doesn't exist in the default config.
```

Comment:
```suggestion
  /// - Parameter key The parameter key of default config.
  /// - Returns The default value of the specified key if the key exists;
  ///  otherwise, nil.
```

Reply:
done